### PR TITLE
Implement Eternatus VMAX ability

### PR DESCRIPTION
**Pull request by luongthomas**
This PR introduces the first version of Eternatus VMAX's ability

The changes here are dependent on 2 other PRs

https://github.com/axpendix/tcgone/pull/29
https://github.com/axpendix/tcgone-engine-contrib/pull/897

### CODE REVIEW

**Comment by luongthomas** [comment]
[ETERNATUS_VMAX_128](https://trello.com/c/byo4O2Iz/562-eternatusvmax128)

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 1]
```diff
@@ -2955,7 +2960,94 @@ public enum DarknessAblaze implements LogicCardInfo {
         weakness F
         bwAbility "Infinity Zone", {
           text "If all of your Pokémon in play are [D], you may now have up to 8 [D] Pokémon on your Bench and can’t play any other type of Pokémon. (When this Ability no longer works, discard your Benched Pokémon until there are 5 left.)"
-          actionA {
+          bwAbility "Infinity Zone", {
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:2963]: `bwAbility "Infinity Zone"` already is there.

Comment (luongthomas) [reply src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:2963]: Thanks, nice catch

**Comment by flagrama** [comment]
Weird timing issue I think. I use Super Scoop Up to pick up my Eternatus VMAX on the bench, then the bench discard selection lets me choose that same Eternatus VMAX, which causes a game error.
game log trace: https://gist.github.com/flagrama/43dbae5134ad7f21b6eef5afbe242ceb

File: src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy [hunk 2]
```diff
@@ -263,6 +263,11 @@ public enum DarknessAblaze implements LogicCardInfo {
     return tcgwars.logic.card.Collection.DARKNESS_ABLAZE;
   }
 
+  @Override
+  public int getCollectionLineNo() {
+    return collectionLineNo;
+  }
+
```
Comment (starg09) [inline src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy:270]: This is causing build to fail. Was removed in axpendix rework of collection numbers iirc

```suggestion
```

**Comment by luongthomas** [comment]
> * [ ]  Removing all non-Dark type Pokemon from the field does not (re)activate Eternal Zone.

Can you provide an example?  I've tested removing all non-darkness type Pokemon and I was able to get the Bench size to increase to 8

**Comment by starg09** [comment]
> [ ] Weird timing issue I think. I use Super Scoop Up to pick up my Eternatus VMAX on the bench, then the bench discard selection lets me choose that same Eternatus VMAX, which causes a game error.

Hmm, weird. In my local instance the Eternatus doesn't show up (a blank space is put in its spot though, inside the select). And I'm forced down to a bench of 4.

**Comment by flagrama** [comment]
> Hmm, weird. In my local instance the Eternatus doesn't show up (a blank space is put in its spot though, inside the select). And I'm forced down to a bench of 4.

Sounds to me like you didn't evolve him and just played him straight from hand?

> Can you provide an example? I've tested removing all non-darkness type Pokemon and I was able to get the Bench size to increase to 8

I removed a non-Dark-type Pokémon from the active spot IIRC.

**Comment by luongthomas** [comment]
Testing reopening

**Comment by starg09** [comment]
> Sounds to me like you didn't evolve him and just played him straight from hand?

@flagrama First time for sure, but I think I also tested it via evolving it and scooping it with SSU specifically.

**Pull request closed without merging**